src/sentry/search/events/datasets/functions.py

```diff
@@ -0,0 +1,359 @@
+    Column(alias="transaction", column="transaction_name", kind=Kind.STRING),
+    Column(alias="timestamp", column="timestamp", kind=Kind.DATE),
+    Column(alias="depth", column="depth", kind=Kind.INTEGER),
+    Column(alias="parent_fingerprint", column="parent_fingerprint", kind=Kind.INTEGER),
+    Column(alias="fingerprint", column="fingerprint", kind=Kind.INTEGER),
+    Column(alias="name", column="name", kind=Kind.STRING),
+    Column(alias="package", column="package", kind=Kind.STRING),
+    Column(alias="path", column="path", kind=Kind.STRING),
+    Column(alias="is_application", column="is_application", kind=Kind.INTEGER),
+    Column(alias="platform.name", column="platform", kind=Kind.STRING),
+    Column(alias="environment", column="environment", kind=Kind.STRING),
+    Column(alias="release", column="release", kind=Kind.STRING),
+    Column(alias="os.name", column="os_name", kind=Kind.STRING),
+    Column(alias="os.version", column="os_version", kind=Kind.STRING),
+    Column(alias="retention_days", column="retention_days", kind=Kind.INTEGER),
+    Column(alias="percentiles", column="percentiles", kind=Kind.DURATION),
+]
+
+COLUMN_MAP = {column.alias: column for column in COLUMNS}
+
+
+class FunctionColumnArg(ColumnArg):  # type: ignore
+    def normalize(
+        self, value: str, params: ParamsType, combinator: Optional[Combinator]
+    ) -> NormalizedArg:
+        column = COLUMN_MAP.get(value)
+
+        # must be a known column or field alias
+        if column is None and value not in {PROJECT_ALIAS, PROJECT_NAME_ALIAS}:
+            raise InvalidFunctionArgument(f"{value} is not a valid column")
+
+        return value
+
+
+class FunctionNumericColumn(NumericColumn):  # type: ignore
+    def _normalize(self, value: str) -> str:
+        column = COLUMN_MAP.get(value)
+
+        if column is None:
+            raise InvalidFunctionArgument(f"{value} is not a valid column")
+
+        if (
+            column.kind == Kind.INTEGER
+            or column.kind == Kind.DURATION
+            or column.kind == Kind.NUMBER
+        ):
+            return column.column
+
+        raise InvalidFunctionArgument(f"{value} is not a numeric column")
+
+    def get_type(self, value: str) -> str:
+        try:
+            return COLUMN_MAP[value].kind.value
+        except KeyError:
+            return Kind.NUMBER.value
+
+
+class FunctionsDatasetConfig(DatasetConfig):
+    non_nullable_keys = {
+        "project.id",
+        "project_id",
+        "transaction_name",
+        "timestamp",
+        "depth",
+        "parent_fingerprint",
+        "fingerprint",
+        "name",
+        "package",
+        "path",
+        "is_application",
+        "platform",
+        "os_name",
+        "os_version",
+        "retention_days",
+        "count",
+        "percentiles",
+        "min",
+        "max",
+        "avg",
+        "sum",
+        "worst",
+        "examples",
```

Comment:
Do these aggregate columns need to be specified in the list of `non_nullable_keys`?

Reply:
probably not, i gave it the ol' copy and paste technique.. i'll rm them